src/lib/backup-storage-availability/backup-storage-availability.ts

```diff
@@ -0,0 +1,122 @@
+import { exec } from 'shelljs';
+import path from 'path';
+import xdgBasedir from 'xdg-basedir';
+import os from 'os';
+import checkDiskSpace from 'check-disk-space';
+import { Confirm } from 'enquirer';
+import { Job } from '../../graphqlTypes';
+
+const oneGiBInBytes = 1024 * 1024 * 1024;
+
+export class BackupStorageAvailability {
+	archiveSize: number;
+
+	constructor( archiveSize: number ) {
+		this.archiveSize = archiveSize;
+	}
+
+	static createFromDbCopyJob( job: Job ): BackupStorageAvailability {
+		const bytesWrittenMeta = job.metadata?.find( meta => meta?.name === 'bytesWritten' );
+		if ( ! bytesWrittenMeta?.value ) {
+			throw new Error( 'Meta not found' );
+		}
+
+		return new BackupStorageAvailability( Number( bytesWrittenMeta.value ) );
+	}
+
+	getDockerStorageAvailable(): number {
+		const kiBLeft = exec( `docker run --rm alpine df -k`, { silent: true } )
+			.grep( /\/dev\/vda1/ )
+			.head( { '-n': 1 } )
+			.replace( /\s+/g, ' ' )
+			.split( ' ' )[ 3 ];
+
+		return Number( kiBLeft ) * 1024;
+	}
+
+	bytesToHuman( bytes: number ) {
```

Comment:
Should we use this util function instead? https://github.com/Automattic/vip-cli/blob/40ea4b18d3b2174db265dbe914b549dfc9a5c422/src/lib/cli/format.ts#L230

Reply:
Probably quite unlikely for users to encounter this, but mentioning it all the same: `bytesToHuman()` can also end up rounding to 0.00GB for small archives resulting in a message `We recommend that you have at least 0.00 GB of free space in your machine...`  It only happens in `validateAndPromptDiskSpaceWarningForBackupImport()` because we use the raw archive size to determine the storage requirement.

I encountered this running an export from my test site which produced an archive that was 2359458 bytes.

Reply:
Sure, let's do that!

However, I'll need a small reimplementation if we want to stick with what Mac and Docker says about GB or it'll introduce confusion. Shouldn't be too hard...